File: 735-asteroid-collision/asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> st=new Stack<>();
        for(int i=0;i<asteroids.length;i++){
            boolean flag=false;
            int val=asteroids[i];
            while(!st.isEmpty() && st.peek()>0 && val<0){
                int k=Math.abs(val);
                if(k==st.peek()){
                    st.pop();
                    flag=true;
                    break;
                }
                if(st.peek()>k){
                    flag=true;
                    break;
                }
                else{
                    st.pop();
                }
            }
            if(!flag){
                st.push(val);
            }
        }
        int[] ans=new int[st.size()];
        int k=0;
        for(int i:st){
            ans[k++]=i;
        }
        return ans;
    }
}

File: 735-asteroid-collision/asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        vector<int> ans;
        stack<int> st;
        for(int i=0;i<asteroids.size();i++){
            bool flag=false;
            int val=asteroids[i];
            while(!st.empty() && st.top()>0 && val<0){
                int k=abs(val);
                if(k==st.top()){
                    st.pop();
                    flag=true;
                    break;
                }
                else if(k>st.top()){
                    st.pop();
                }
                else{
                    flag=true;
                    break;
                }
            }
            if(!flag){
                st.push(val);
            }
        }
        while (!st.empty()) {
            ans.push_back(st.top());
            st.pop();
        }

        reverse(ans.begin(), ans.end());
        return ans;
    }
};